polkadot-protocol/protocol-components/accounts/account-data-structure.md

```diff
@@ -0,0 +1,132 @@
+
+The providers counter must be greater than zero for the consumers counter to be incremented.
+
+### Providers Reference Counters
+
+The `providers` reference counter indicates whether an account is ready to be depended upon. For example, it is incremented when a new account is created with a balance greater than the existential deposit.
+
+The `providers` reference counter prevents Substrate pallets from storing data about an account until the account is active (`providers` > 0)
+
+### Consumers Reference Counters
+
+The `consumers` reference counter ensures that Substrate pallets don't remove an account until all associated data across pallets is deleted (that is, when `consumers` == 0). 
+
+It holds users accountable for the data they store on-chain. If users wish to delete their accounts and reclaim their existential deposit, they must first remove all data stored across all on-chain pallets to decrement the `consumers` counter.
+
+### Sufficients Reference Counter
+
+The `sufficients` reference counter indicates whether an account is self-sufficient and can exist independently. 
+
+For example, in the [`assets`](https://paritytech.github.io/polkadot-sdk/master/pallet_assets/index.html) pallet, an account may hold a sufficient quantity of specific assets without needing to maintain a native account balance.
+
+### Account Deactivation 
+
+Pallets also include cleanup functions that decrement the `providers` reference counter, marking the account as deactivated within the pallet-managed scope. 
+
+Once both the `providers` and `consumers` reference counters reach zero, the account is considered deactivated by all on-chain pallets.
+
+### Updating Counters
+
+Runtime developers can manage these counters using the `inc_consumers()`, `dec_consumers()`, `inc_providers()`, `dec_providers()`, `inc_sufficients()`, and `dec_sufficients()` methods provided by the `frame-system` pallet. 
+
+For every increment of a counter during an account's lifecycle, a corresponding decrement should be called to ensure proper management of the account state.
+
+There are also three query functions to ease usage on these counters:
+
+- `can_inc_consumer()` to check if an account is ready to be used (`providers` > 0).
+- `can_dec_provider()` to check if an account is no longer referenced in runtime whatsoever (`consumers` == 0) before decrementing providers to 0.
+- `is_provider_required()` to check if an account has outstanding consumer references (`consumers` > 0).
+
+!!! note
+    See [frame-system API](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/struct.Pallet.html) for more details.
+
+## AccountData Trait and Implementation
+
+The `AccountInfo` can be any structure, provided it meets the requirements of the associated type `AccountData` trait bound defined in the [`frame-system::pallet::Config`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/trait.Config.html) trait. 
```

Comment:
Link AccountInfo via Polkadot SDK Docs

Reply:
I linked AccountInfo to frame_system::AccountInfo , tell me if this is the correct linkl